build.zig

```diff
@@ -44,28 +44,31 @@ pub fn build(b: *std.Build) !void {
     }
 
     // TODO: change to addExecutable with entry disabled when we update to zig 0.12.0.
-    const lib = b.addSharedLibrary(.{
+    const lib = b.addExecutable(.{
         .name = "cart",
-        .root_source_file = .{ .path = "platform/host.zig" },
-        .target = .{ .cpu_arch = .wasm32, .os_tag = .freestanding },
+        .root_source_file = b.path("platform/host.zig"),
+        .target = b.resolveTargetQuery(.{
+            .cpu_arch = .wasm32,
+            .os_tag = .freestanding,
+        }),
         .optimize = optimize,
     });
```

Comment:
I'm getting this error:

`error: wasm-ld: entry symbol not defined (pass --no-entry to suppress): _start`

I suspect this has to do with the @bhansconnect's comment, but not sure what I need to change.

Reply:
Got this working with `lib.entry = .disabled;` If we want to revert back to `addSharedLibrary`, this will take some more tweaking.

Reply:
Cool. @bhansconnect you may be interested to know. 😀